### PR TITLE
Q2d methods

### PR DESCRIPTION
**Pull request by ring630**
create_rectangle()
asisgn_single_signal_line

### CODE REVIEW

File: pyaedt/q3d.py [hunk 1]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
         )
+
+    def create_rectangle(self, position, dimension_list, name="", matname=""):
+        """
+        Create a rectangle.
+
+        Parameters
+        ----------
+        position : list
+            List of  [x, y] coordinates for the starting point of rectangle
```
Comment (PipKat) [inline pyaedt/q3d.py:566]:
```suggestion
            List of [x, y] coordinates for the starting point of rectangle
```

Comment (PipKat) [inline pyaedt/q3d.py:566]:
```suggestion
            List of [x, y] coordinates for the starting point of the rectangle.
```

File: pyaedt/q3d.py [hunk 2]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+        Assign conductor type to sheets.
+
+        Parameters
+        ----------
+        name : str
+            Name of the conductor.
+        target_objects : list
+            List of Object3D.
+        solve_option : str
+            Choose among "SolveInside", "SolveOnBoundary" or Automatic. Default is SolveInside.
```
Comment (PipKat) [inline pyaedt/q3d.py:595]:
```suggestion
            Method for solving. Options are ``"SolveInside"``, ``"SolveOnBoundary"`` or ``"Automatic"``. The default is ``"SolveInside"``.
```

File: pyaedt/q3d.py [hunk 3]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+        """
+        Assign conductor type to sheets.
+
+        Parameters
+        ----------
+        name : str
+            Name of the conductor.
+        target_objects : list
+            List of Object3D.
+        solve_option : str
```
Comment (PipKat) [inline pyaedt/q3d.py:594]:
```suggestion
        solve_option : str, optional
```

Comment (PipKat) [reply pyaedt/q3d.py:594]: Please reorder the parameters to match the order in which they are referenced in the function expression.

File: pyaedt/q3d.py [hunk 4]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+            Choose among "SolveInside", "SolveOnBoundary" or Automatic. Default is SolveInside.
+        thickness : float
```
Comment (PipKat) [inline pyaedt/q3d.py:596]:
```suggestion
        thickness : float, optional
```

File: pyaedt/q3d.py [hunk 5]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+            Conductor thickness. When not specified, the default conductor thickness is obtained by dividing the
```
Comment (PipKat) [inline pyaedt/q3d.py:597]:
```suggestion
            Conductor thickness. The default is ``None``, in which case the conductor thickness is obtained by dividing the
```

File: pyaedt/q3d.py [hunk 6]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+            conductor's area by its perimeter (A/p). If multiple conductors are selected, the average conductor
+            thickness is used for the default value.
```
Comment (PipKat) [inline pyaedt/q3d.py:599]:
```suggestion
            thickness is used.
```

File: pyaedt/q3d.py [hunk 7]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+        unit : str
+            Thickness unit.
```
Comment (PipKat) [inline pyaedt/q3d.py:601]:
```suggestion
            Thickness unit. The default is ``"um"``.
```

File: pyaedt/q3d.py [hunk 8]
```diff
@@ -554,3 +555,116 @@ def __init__(
...
+        Returns
+        -------
+        None.
```
Comment (PipKat) [inline pyaedt/q3d.py:604]: I don't think we include a "Returns" section is nothing is returned. Someone else can confirm. I'll review all the other docstrings added after the merge.

**Review by maxcapodi78 — APPROVED** [review]

File: pyaedt/q3d.py [hunk 9]
```diff
@@ -554,3 +555,111 @@ def __init__(
             close_on_exit,
             student_version,
         )
+
+    def create_rectangle(self, position, dimension_list, name="", matname=""):
```
[the remark this answers is not in the document]
Comment (ring630) [reply pyaedt/q3d.py:559]: It is ok to use q2d.modeler.primitives.create_rectangle, but the user must have the prior knowledge create_rectangle is under q2d.modeler, primitives. In addition, q2d.modeler.primitives.create_rectangle requires Position(x, y, z) as input, which belongs to modeler. q2d.create_rectangle makes PyAEDT easier for people to use.

**Pull request merged**